Add docs for failed_data_files_count (#8341)
apache/iceberg: docs/spark-procedures.md

| `rewritten_data_files_count` | int | Number of data which were re-written by this command |
| `added_data_files_count`     | int | Number of new data files which were written by this command |
| `rewritten_bytes_count`      | long | Number of bytes which were written by this command |

> | `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true |

Reviewer: ```suggestion
| `failed_data_files_count`    | int | Number of data files that failed to be rewritten when `partial-progress.enabled` is true |
```

> | `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true and any rewrites failed |

Reviewer 2: ```suggestion
| `failed_data_files_count`    | int | Number of data files that failed to write if `partial-progress.enabled` is true |
```

I don't think the last bit is necessary if the count is 0 if none failed.
